### DISCUSSION TITLE
Can DAFoam do multipoint aerostructural optimization

### DISCUSSION THREAD

**Discussion by Ryupuriwatt** [mdolab/dafoam#713] [General]
![image](https://github.com/user-attachments/assets/2c2957b7-9184-42e5-a4e5-2a8961889c5d)
![image](https://github.com/user-attachments/assets/7a7edd2b-4661-4b45-b015-0c6d60426524)
Hello profressor, This is line of code that has problem. I try to do multipoint aerostructural optimization but not sure available or not. I know that multipoint aerodynamic optimization is possible. Another question is that is it possible to do nonlinear structural analysis for tacs for newest version of dafoam or we can just upgrade the version of tacs to the newest version to use nonlinear solver.

**Comment by friedenhe** [comment]
Please use this as the template for multi-point: https://github.com/DAFoam/tutorials/blob/main/NACA0012_Airfoil/multipoint/runScript.py. The way you called mphys_add_funcs is not correct.
